src/main/java/org/semux/core/BlockHeader.java

```diff
@@ -71,7 +71,7 @@ public BlockHeader(long number, byte[] coinbase, byte[] prevHash, long timestamp
         enc.writeBytes(stateRoot);
         enc.writeBytes(data);
         this.encoded = enc.toBytes();
-        this.hash = Hash.h256(encoded);
+        this.hash = Hash.h256_s(encoded, null);
```

Comment:
why does this need changing?  the block header algo should be unchanged, else we'd need to use forkflag.